packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js

```diff
@@ -97,8 +98,15 @@ export class FieldSelectContainer extends PureComponent {
         return sortedOptions;
     }
 
+    isSelectDisabled() {
+        const { selectOptions } = this.props;
+        return selectOptions.length === 0;
+    }
+
     handleSelectExpand() {
-        this.setState(({ isSelectExpanded }) => ({ isSelectExpanded: !isSelectExpanded }));
+        if (!this.isSelectDisabled()) {
+            this.setState(({ isSelectExpanded }) => ({ isSelectExpanded: !isSelectExpanded }));
+        }
```

Comment:
Prefer inversing if statements in order to have early returns, rather than having logic within `if` statements.